docs/docs-beta/docs/guides/automation/asset-sensors.md

```diff
@@ -1,10 +1,80 @@
+sidebar_position: 30
 ---
 
-### Basic Asset Sensor Example
+Asset sensors in Dagster provide a powerful mechanism for monitoring asset materializations and triggering downstream computations or notifications based on those events.
+
+There are many use cases for asset sensors, however the most common is to trigger a job when an asset is materialized in a different job or code location, or to provide custom logic to determine when to trigger a run following an asset materialization.
+
+### Cross-job and cross-code location dependencies
+
+Asset Sensors enable dependencies across different jobs and even different code locations. This flexibility allows for more modular and decoupled workflows.
+
+
+```mermaid
+graph LR;
+
+AssetToWatch --> AssetSensor;
+AssetSensor --> Job;
+Job --> Asset1;
+Job --> Asset2;
+
+subgraph CodeLocationA
+    AssetToWatch
+end
+
+subgraph CodeLocationB
+    AssetSensor
+    Job
+    Asset1
+    Asset2
+end
+
+```
+
+Here's a minimal example of an asset sensor that triggers a job when an asset is materialized. The `daily_sales_data` asset is in the same code location for this example, but the same pattern can be applied to assets in different code locations.
 
 <CodeExample filePath="guides/automation/simple-asset-sensor-example.py" language="python" title="Simple Asset Sensor Example" />
 
-This Asset Sensor will trigger a run of `my_job` whenever the `asset_to_watch` asset is materialized.
+### Custom evaluation logic
+
+The evaluation function of an asset sensor can be customized to include custom logic for determining when to trigger a run. This allows for fine-grained control over the conditions under which downstream jobs are executed.
+
+```mermaid
+stateDiagram-v2
+    direction LR
+
+
+
+    [*] --> AssetMaterialized
+    AssetMaterialized --> [*]
+
+    AssetMaterialized --> UserEvaluationFunction
+    UserEvaluationFunction --> RunRequest
+    UserEvaluationFunction --> SkipReason
+    SkipReason --> [*]
+    RunRequest --> [*]
+
+    class UserEvaluationFunction userDefined
+    classDef userDefined fill: lightblue
+```
+
+In this example, the `@asset_sensor` decorator allows us to define a custom evaluation function that returns a `RunRequest` object when the asset is materialized and certain metadata is present,
+otherwise it skips the run.
+
+<CodeExample filePath="guides/automation/asset-sensor-custom-eval.py" language="python" title="Asset Sensor with Custom Evaluation Logic" />
+
+## Trigger a job with configuration
+
+By providing a configuration to the `RunRequest` object, you can trigger a job with a specific configuration. This is useful when you want to trigger a job with custom parameters based on custom logic you define. For example, you might use a sensor to trigger a job when an asset is materialized, but also pass metadata about that materialization to the job. 
+
+<CodeExample filePath="guides/automation/asset-sensor-with-config.py" language="python" title="Asset Sensor with Config" />
```

Comment:
Call out in what sense?

Reply:
In that we specify "weekly_report" as the key to the `ops` parameter because an op of that name underlies our asset, and this is a byproduct of how assets are implemented.

But ideally this will just go away.

Reply:
ah got it. With guides I tend to try not to explain too much, even though there's much that needs explaining because otherwise it can turn into a Concept page. I think the right place to deal with this, in order of preference is 
1. the product, like you said
2. API docs 
3. Concept page